enhancements/authentication/direct-external-oidc-provider.md

```diff
@@ -212,6 +242,76 @@ These changes have already been implemented, and the initial PR for them can be
+- The post-start hook that checks for oauth-apiserver connectivity will be skipped if the `Authentication` resource `.spec.type` is set to `OIDC`. This will prevent logs in the kube-apiserver associated with not being able to connect to the oauth-apiserver, which we know should not be running when OIDC is enabled.
+
+##### Changes to the cluster-kube-apiserver-operator
+
+When authentication type is set to OIDC, the `authorization.openshift.io/RestrictSubjectBindings` and `authorization.openshift.io/ValidateRoleBindingRestriction` admission plugins will be disabled.
+
+This will be done through updates to the appropriate config observers to update the `KubeAPIServerConfig.apiServerArguments` map to:
+
+- Remove the `authorization.openshift.io/RestrictSubjectBindings` and `authorization.openshift.io/ValidateRoleBindingRestriction` admission plugins from the `--enable-admission-plugins` argument
+- Add the `authorization.openshift.io/RestrictSubjectBindings` and `authorization.openshift.io/ValidateRoleBindingRestriction` admission plugins to the `--disable-admission-plugins` argument
+
+##### Changes to the cluster-authentication-operator
+
+A couple changes will need to be made to the cluster-authentication-operator. Mainly:
+- Ownership/Management of the `rolebindingrestrictions.authorization.openshift.io` CustomResourceDefinition
+- Blocking OIDC configuration due to existence of `RoleBindingRestriction` resources and communicating it via the `Authentication` resources new `OIDCConfig` status field
+
+To support the need to remove the `rolebindingrestrictions.authorization.openshift.io` CustomResourceDefinition when configuring the authentication type as OIDC, the cluster-authentication-operator will be updated to manage this CRD.
+
+This will mean vendoring the generated CRD manifests as outlined in https://github.com/openshift/api/tree/master?tab=readme-ov-file#vendoring-generated-manifests-into-other-repositories and adding a new controller to manage the CRD.
+
+Managing the CRD will consist of ensuring that the CRD is present on the cluster, and matches the desired manifest, when the authentication type is _not_ OIDC, and ensuring the CRD is not present present on the cluster when the authentication type _is_ OIDC and OIDC configuration has been successfully rolled out. 
+
+Additionally, the CAO will be updated to block OIDC configuration on existence of `RoleBindingRestriction` resources. If `RoleBindingRestriction` resources are found,
+the Authentication CR's `OIDCConfig` status field will be updated to contain the following conditions:
+
+- Condition: `Progressing`, Status: `False`, Reason: `Blocked`, Message: `OIDC configuration blocked: RoleBindingRestriction resources found`
```

Comment:
Can't force progressing to false.  it'll go progressing for other reasons.  Instead use Degraded.

Reply:
This is for a non-cluster operator related resource. This api field doesn't currently exist and therefore nothing is setting it to progressing. I'm fine with changing this to degraded if you think that makes the most sense for this case, but we were trying to be cautious of setting degraded unless the cluster is actually in a broken state.